### PR TITLE
Feature/gym

### PR DESCRIPTION
**Pull request by vanpelt**
This adds support for logging gym videos and a wandb logger for ray.  I put them in the same PR because they're both serving RL environments.  Theres a new `monitor_gym` keyword arg to wandb.init or users can call `wandb.gym.monitor`.  There's also a new wandb.ray module with a WandbLogger.

This also changes the way tensorboard events are logged.  Instead of passing step to wandb.log we keep track of the step in the tf logger and commit only when it increases within a namespace.  This will not log errors if step decreases but nicely handles the case of multiple step counters across namespaces which I was seeing in github RL repos.

### CODE REVIEW

**Comment by vanpelt** [comment]
@shawnlewis @raubitsj want to get this in.  What do we think about the monitor_gym keyword arg to init?  I'm a little wary of throwing more shit in there...

File: wandb/tensorboard/__init__.py [hunk 1]
```diff
@@ -141,14 +142,34 @@ def new_csfw_func(*args, **kwargs):
             [TENSORBOARD_C_MODULE, "create_summary_file_writer"])
 
 
+STEPS = {"": {"step": 0}}
+
+
 def log(tf_summary_str, history=None, **kwargs):
-    namespace = kwargs.get("namespace")
+    """Logs a tfsummary to wandb"""
+    global steps
```
Comment (raubitsj) [inline wandb/tensorboard/__init__.py]: uppercase steps?

File: wandb/tensorboard/__init__.py [hunk 2]
```diff
@@ -141,14 +142,34 @@ def new_csfw_func(*args, **kwargs):
...
         del kwargs["namespace"]
+    # To handle multiple global_steps, we keep track of them here instead of the global log
+    last_step = STEPS.get(namespace, {"step": 0})
+    cur_step = kwargs.get("step", 0)
+    if last_step["step"] < cur_step:
+        kwargs["commit"] = True
+    else:
+        kwargs["commit"] = False
+    STEPS[namespace] = {"step": cur_step}
+    print(namespace, last_step, cur_step, kwargs["commit"])
```
Comment (raubitsj) [inline wandb/tensorboard/__init__.py]: remove debug print

**Comment by raubitsj** [comment]
> @shawnlewis @raubitsj want to get this in. What do we think about the monitor_gym keyword arg to init? I'm a little wary of throwing more shit in there...

Seems reasonable as it is a global operation

**Pull request merged**